### ISSUE TITLE
Proposal to add C++ code editor in Engine as Addon

### ISSUE THREAD

**Issue by Ughuuu** [blazium-engine/blazium#75]
The idea is to keep this as easy as possible. The workflow would be similar to this: https://libriscv.no/docs/godot/rustprogram

**Rust Usage Workflow**:
When creating a Script you have option to create Rust script file:
<img width="484" alt="image" src="https://github.com/user-attachments/assets/16991d36-7aa5-4965-bd25-642e7cb10530">

- Create file (just normal C++ file)
![](https://libriscv.no/assets/images/new-script-24f64f5a40fd8380f3f28fe9ab8715d9.png)
- Edit c++ files in the editor

<img width="954" alt="image" src="https://github.com/user-attachments/assets/7104dc27-5116-45a5-b711-c6e3a40da1ff">

- Compile them

Since this would be generic, it would probably support GDExtension. So you would how to be able to write c++ code and build it, either when using regular CMake projects, SConstruct projects (when doing GDExtension addon), or something else.

## Autocomplete

For this the code needs to be processed to offer autocomplete options

## Build

For this there can be a build option, similar to how C# has. Also, if not, when Saving files and/or when running project, build the code to a GDExtension or DLL (custom lib).

### Compiler

As for building, we could support either setting compiler path and using that, or using local Docker and use that (send files to the Docker and get the binary output and save it).

**Comment by drcucaracha** [comment]
Hi, maybe this is off topic, but wouldn't it be easier to write everything in gdscript and before creating the executable use something like the plugin (https://github.com/Lcbx/GdScript2All) that translates the code and creates the executable with the selected code?

**Comment by Ughuuu** [comment]
Interesting, will read that and see how it is useful and where it could fall into all this.

**Comment by Ughuuu** [comment]
Yeah, this needs the Godot 4.4 to work (with godot-sandbox module)

**Comment by ShalokShalom** [comment]
There is also https://github.com/Jenova-Framework/J.E.N.O.V.A/

It needs some ability to read the meta data of DWARF files (especially around pure types) to be available on macOS and Linux as well. 

Its possible to implement a workaround with a slower interpreter, although I think its better and easier to implement the DWARF thing.

**Comment by Bioblaze** [comment]
closing this since jenova, and sandbox are both going to handle this for us.
